[14.0][MIG] stock_picking_supplier_ref: Migration to 14.0

Reyes4711-S73 opened the pull request:
Standard migration to 14.0

Commit:
[ADD] TH181130/0011 stock_picking_supplier_ref

Commit:
[FIX] Add OCA author

Commit:
[FIX] Flake8

Commit:
[FIX] Add module description

Reyes4711-S73 committed:
[10.0][IMP] stock_picking_supplier_ref: black, isort, prettier

Reyes4711-S73 committed:
[14.0][MIG] stock_picking_supplier_ref: Migration to 14.0

flotho commented on stock_picking_supplier_ref/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1 @@
+This module adds the field supplier reference in pickings.
```
Which reference are we talking about

flotho commented on stock_picking_supplier_ref/models/stock_picking.py:
```diff
@@ -0,0 +1,10 @@
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    reference = fields.Char(string="Supplier Reference")
```
IMHO reference is too more "open" definition. I mean, are we talking about the reference of the supplier_reference available in the PO, are we talking about the picking reference of the supplier ?
I would recommend to make some small changes to better specify the use case.

flotho reviewed (changes requested):
Hi,

Thanks for your work.
I think that small changes could easily improve your contribution.

Regards

Reyes4711-S73 commented:
@flotho In this PR we are doing only the standard migration of this module, not doing some improvements. To do this, we need to create another PR with all the improvements to do.
Besides, changing the name of this field maybe do some problems with people are using this module. In the PR of improvements, you must do a script which does the change of the name of field to the new name.

rousseldenis commented:
> @flotho In this PR we are doing only the standard migration of this module, not doing some improvements. To do this, we need to create another PR with all the improvements to do. Besides, changing the name of this field maybe do some problems with people are using this module. In the PR of improvements, you must do a script which does the change of the name of field to the new name.

@Reyes4711-S73 You can do for sure changes in this PR. For reference name, I agree with @flotho. To support existing installations, you should provide a migration script.

Reyes4711-S73 commented:
@rousseldenis @flotho In a few days I will do this changes

Reyes4711-S73 committed:
[14.0][IMP] stock_picking_supplier_ref: module improvements

Adds improvements in the module to simplify the understanding of the module operation

Reyes4711-S73 commented:
@rousseldenis and @flotho Added suggested improvements. If is all correct, could you merge this PR?

rousseldenis commented:
/ocabot migration stock_picking_supplier_ref

Reyes4711-S73 commented:
@rousseldenis Could you merge this PR?

rousseldenis commented:
@flotho Could you update your review?

@Reyes4711-S73 Could you squash your two last commits (as the add of openupgradelib is linked to the improvement)?

flotho reviewed (approved):
LGTM, thanks for the update

Reyes4711-S73 commented:
@rousseldenis Squashed. Please, could you merge this PR now?

rousseldenis commented:
/ocabot merge patch
